### PR TITLE
Ledger WebHID What's New popup

### PR DESCRIPTION
**Pull request by danjm**
Adds a What's New notification for Ledger Live users, instructing them that they can now connect with WebHID

Screenshot from 2021-10-27 15-59-54

### CODE REVIEW

**Review by darkwing — CHANGES_REQUESTED** [review]
One tiny bit of feedback

**Review by darkwing — DISMISSED** [review]

**Review by jj-crypto — DISMISSED** [review]

**Review by jj-crypto — DISMISSED** [review]

File: app/_locales/en/messages.json [hunk 1]
```diff
@@ -1648,6 +1648,22 @@
...
+  "notifications8ActionText": {
+    "message": "Go to Advanced Settings",
+    "description": "Description on an action button that appears in the What's New popup. Tells the user that if they click it, they will go to our Advanced Settings page."
+  },
+  "notifications8DescriptionOne": {
+    "message": "As of MetaMask v10.4.0, you no longer need Ledger Live to connect your Ledger device to MetaMask.",
+    "description": "Description of a notification in the 'See What's New' popup. Describes changes for how Ledger Live is no longer needed to connect the device."
+  },
+  "notifications8DescriptionTwo": {
+    "message": "You can go to the Advanced tab of settings and switch the 'Preferred Ledger Connection Type' to 'WebHID'.",
```
[the remark this answers is not in the document]
Comment (mcmire) [reply app/_locales/en/messages.json:1675]: Other than this, what do you think about dropping the "You can go to..." and just saying "Go to..."?

[the remark this answers is not in the document]
Comment (danjm) [reply app/_locales/en/messages.json]: There is a help article linked from the advanced tab in settings

The wording has been updated in 0fb7ff3ec to reflect the above suggestion and a suggestion from marketing

**Review by tmashuang — DISMISSED** [review]
lgtm the notification shows with newly, and preexisting, connected ledger accounts via ledger live.

**Review by jpuri — DISMISSED** [review]
LGTM

File: app/_locales/en/messages.json [hunk 2]
```diff
@@ -1663,6 +1663,22 @@
...
+  "notifications8DescriptionOne": {
+    "message": "As of MetaMask v10.4.0, you no longer need Ledger Live to connect your Ledger device to MetaMask.",
+    "description": "Description of a notification in the 'See What's New' popup. Describes changes for how Ledger Live is no longer needed to connect the device."
+  },
+  "notifications8DescriptionTwo": {
+    "message": "For an easier and more stable ledger experience, go to the Advanced tab of settings and switch the 'Preferred Ledger Connection Type' to 'WebHID'.",
+    "description": "Description of a notification in the 'See What's New' popup. Describes how the user can turn off the Ledger Live setting."
+  },
+  "notifications8Title": {
+    "message": "Ledger connection improvement.",
```
Comment (danjm) [inline app/_locales/en/messages.json:1679]:
```suggestion
    "message": "Ledger connection improvement",
```

File: ui/selectors/selectors.js [hunk 3]
```diff
@@ -570,7 +570,13 @@ export function getShowWhatsNewPopup(state) {
  * @param {Object} state
  * @returns {Object}
  */
-function getAllowedNotificationIds() {
+function getAllowedNotificationIds(state) {
+  const currentKeyring = getCurrentKeyring(state);
+  const currentKeyringIsLedger = currentKeyring?.type === KEYRING_TYPES.LEDGER;
+  const supportsWebHid = Boolean(window.navigator.hid);
```
[the remark this answers is not in the document]
Comment (danjm) [reply ui/selectors/selectors.js:581]: Apparently we may stop using the `in` operator soon: https://github.com/MetaMask/eth-ledger-bridge-keyring/pull/106#discussion_r718939285

So I will change this to an `undefined` check

File: ui/selectors/selectors.js [hunk 4]
```diff
@@ -575,7 +575,13 @@ export function getShowWhatsNewPopup(state) {
...
```
Comment (danjm) [inline ui/selectors/selectors.js:581]:
```suggestion
  const supportsWebHid = window.navigator.hid !== undefined;
```

**Review by mcmire — APPROVED** [review]
Updates look good!

**Review by darkwing — APPROVED** [review]

**Review by ryanml — APPROVED** [review]

**Pull request merged**